XamarinCommunityToolkit/Views/GravatarImageSource/iOS/GravatarImageSourceHandler.ios.cs

```diff
@@ -0,0 +1,33 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Foundation;
+using UIKit;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	public class GravatarImageSourceHandler : IImageSourceHandler
+	{
+		public async Task<UIImage> LoadImageAsync(ImageSource imagesource, CancellationToken cancelationToken = default, float scale = 1)
+		{
+			var fileInfo = await GravatarHandlerUtil.Load(imagesource, scale, GetCacheDirectory()).ConfigureAwait(false);
+
+			if (fileInfo?.Exists ?? false)
+				return UIImage.FromFile(fileInfo.FullName);
+
+			return null;
+		}
+
+		static string GetCacheDirectory()
```

Comment:
If I'm not wrong this method is the same as .macOS. Can we avoid code duplication?

Reply:
it is... I'll look at how I can share it

Reply:
You can create a file with extension `.ios.macos.cs`

Reply:
Yeah. I ended up doing that. Wasn't a huge fan of that as each platform has its own folder and only that method can be shared since Mac uses an NSImage and iOS uses UIImage

Reply:
@dansiegel, feel free to disagree with my ideas! My point for it is just to avoid code duplication hehe. In the future, if Apple changes the API we can move back this method as it's today.

Reply:
Trust me I agree it needed to be deduplicated. Just didn't have great ideas for it.

Reply:
If I have some insight I'll share with you